lib/api/core/plugins/pluginsManager.js

```diff
@@ -421,6 +415,8 @@ class PluginsManager {
             .then(request => cb(null, request))
             .catch(error => cb(error));
         } else {
+          // Do not throw now because we didn't know if the pipe use a callback
```

Comment:
Why ?
I thought this "else" condition was only raise if the pipe method is not a Promise... so we should throw an error here.

Reply:
If I throw directly here, there is no way to distinguish between a pipe using the `callback` parameter and returning `undefined` and a pipe that is using Promise rather than callback and that do not return a valid Promise.

Reply:
Ah, you mean that the pipe method can be not only a Promise, but also a function with a callback parameter ?
But in that case, why do we set an error message with `Plugin pipe must return a valid promise` ?

Reply:
Yes that right. 
```
myPipe(request, callback) {
  callback(null, request);
}
// or 
async myPipe(request) {
  return request;
}
```
I'm not sure to fully remember but I think it's because we want users to use Promise instead of callback in pipes.

Reply:
well, we have 2 choices:

* either we want to force plugin devs to use Promises or Async/Await... and we should throw the error immediately (but it is a breaking change, so it must wait for V2)

* or we want to authorize the usage of callbacks, and this error message has no sense... in that case, we have no other choice than waiting for timeout and throw the generic `Timeout error` message.

Reply:
Ok, I will leave the generic error message